Modules/_datetimemodule.c

```diff
@@ -1873,6 +1873,11 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
     pnew = PyBytes_AsString(newfmt);
     usednew = 0;
 
+    PyObject *strftime = _PyImport_GetModuleAttrString("time", "strftime");
+    if (strftime == NULL) {
+        goto Done;
+    }
```

Comment:
This needs to be moved up above the first `goto`; for example, L1867 will now trigger a `Py_DECREF(strftime)` on an uninitialised pointer.